### PR TITLE
Feature/ims scf

### PR DESCRIPTION
**Pull request by YoulongXia-NOAA**
## Description

Add a functionality to read IMS snow cover fraction (SCF) with native grib2 format and then convert SCF data to ioda-converters NETCDF format.

### Issue(s) addressed

Link the issues to be closed with this PR
- fixes #<#313> - Initial implementation of IODA-converter for IMS data 

## Dependencies

No dependencies

## Impact

No

### CODE REVIEW

**Comment by YoulongXia-NOAA** [comment]
As pygrib is first use in this python code so that Cory and I made optional for ctest.

File: README.md
Comment (CoryMartin-NOAA) [inline README.md:160]: Missing vs mask is maybe not the best description, maybe say that maskout means "to not write out missing values"

File: ioda_envr1
Comment (CoryMartin-NOAA) [inline ioda_envr1:1]: Remove this file from the repository

File: src/land/ims_scf2ioda.py.in
Comment (CoryMartin-NOAA) [inline src/land/ims_scf2ioda.py.in:75]: QC value is always 1? not 0?

**Review by CoryMartin-NOAA — CHANGES_REQUESTED** [review]
Few small changes; remove the env file, can we add the GRIB2 file to git-LFS?, clarify the QC value is correct, and make sure the usage string is clear.

File: ioda_envr1
Comment (YoulongXia-NOAA) [reply ioda_envr1]: removed.

File: README.md
Comment (YoulongXia-NOAA) [reply README.md]: Corrected.

File: src/land/ims_scf2ioda.py.in
Comment (YoulongXia-NOAA) [reply src/land/ims_scf2ioda.py.in]: Changed to 0 and re-do ctest and now it works.

**Comment by srherbener** [comment]
@YoulongXia-NOAA, @markjolah is on parental leave right now so you will need to select another reviewer. Thanks!

File: src/land/ims_scf2ioda.py
Comment (srherbener) [inline src/land/ims_scf2ioda.py:1]: Please change the interpreter to python3 (`#!/usr/bin/env python3`). We want that declaration to be specific because we are phasing out support for python2. Thanks!

**Review by srherbener — COMMENTED** [review]
I got this to work on my iMac by using `-DUSE_PYGRIB=1` on the ecbuild/cmake command line and the `test_iodaconv_ims_scf` test passed. Is this the way you intended the pygrib feature to be enabled?

**Comment by srherbener** [comment]
@YoulongXia-NOAA I guess I volunteered myself to review. If you have someone else in mind, that's okay to add them as well.

**Comment by srherbener** [comment]
@YoulongXia-NOAA, @CoryMartin-NOAA the python installation for the testing likely doesn't have pygrib installed. Do you think that GRIB is prolific enough to warrant making pygrib a standard package for JEDI?

**Comment by YoulongXia-NOAA** [comment]
Stephen, thank you for volunteering to review the code. That is great. I did not have another one as a reviewer.

**Comment by YoulongXia-NOAA** [comment]
Stephen, I have tested it on Hera for ctest, and it works well. My response for your question is yes.  @CoryMartin-NOAA, what do you think?

File: src/land/ims_scf2ioda.py
Comment (YoulongXia-NOAA) [reply src/land/ims_scf2ioda.py]: Corrected.

**Comment by CoryMartin-NOAA** [comment]
GRIB is prolific for meteorology, but not so much for observations (really just model post processed gridded output). So I don't think it needs to be in the JEDI stack for research/development, but operational centers will all have access to it.

**Comment by srherbener** [comment]
@YoulongXia-NOAA, @CoryMartin-NOAA thanks for the feedback. We'll not put pygrib into jedi-stack for now. We can always change that later if the need arises.

The conflicts need to be resolved before we can merge this. If you want, I can help with resolving the conflicts. Thanks!

**Comment by YoulongXia-NOAA** [comment]
@srherbener, yes, please provide the help with resolving the conflicts, and then merge it to develop branch so that I can work for next land variable and product. @CoryMartin-NOAA , I may need to have a meeting with you and instruct me how to resolve the conflicts.

**Comment by YoulongXia-NOAA** [comment]
@srherbener @CoryMartin-NOAA, as Steven showed me today, I have modified  test/CMakeLists.txt to resolve the conflicts. Now I add, commit and push to this feature branch. Steven, please merge it to develop branch. Thanks for both of you!

**Review by CoryMartin-NOAA — APPROVED** [review]

**Review by srherbener — APPROVED** [review]
Looks great. Thanks for fixing the conflicts!

**Pull request merged**
